frontend/src/metabase/query_builder/reducers.js

```diff
@@ -161,16 +160,10 @@ export const uiControls = handleActions(
         isShowingDataReference: !state.isShowingDataReference,
       }),
     },
-    [POP_DATA_REFERENCE_STACK]: {
+    [SET_DATA_REFERENCE_STACK]: {
       next: (state, { payload }) => ({
         ...state,
-        dataReferenceStack: state.dataReferenceStack.slice(0, -1),
```

Comment:
The bug happens because `state.dataReferenceStack` is null when the querybuilder is first initialized.
`getDataReferenceStack(getState())` solves this issue, but `getDataReferenceStack` relies on having access to the root of the redux state. So I've moved this logic [here](https://github.com/metabase/metabase/pull/26049/files#diff-b7bcfe152779a8a85d765f1f5b7570eb1283fd05975ad1473d7dd55135aa80abR39)